tests/test_functional.py

```diff
@@ -35,15 +35,55 @@ def test_home_page(self, testapp, db):
         assert "Evidence dárců ČČK Frýdek-Místek" in res
         assert res.status_code == 200
 
-    def test_pages_401(self, testapp, db):
+
+class TestErrorInterface:
+    """Test 404 and 401 pages"""
+
+    values_404 = [
+        5005165649,
+        5055172826,
+        0,
+        "hello",
+        "aa",
+    ]
+
+    endpoints_keys_404 = (
+        ("donor.detail", "rc"),
+        ("donor.award_prep", "medal_slug"),
+        ("batch.batch_detail", "id"),
+    )
+
+    testcases_404 = {}
+    for endpoint, key in endpoints_keys_404:
+        for value in values_404:
+            testname = f"{endpoint}-{value}"
+            testcases_404[testname] = {
+                "endpoint": endpoint,
+                "kwargs": {key: value},
+            }
+
+    @pytest.mark.parametrize("case_name", testcases_404)
+    def test_dynamic_urls_404(self, user, testapp, case_name):
+        login(user, testapp)
+        case = self.testcases_404[case_name]
+        res = testapp.get(url_for(f"{case['endpoint']}", **case["kwargs"]), status=404)
```

Comment:
There is no need to use formatted string if the only value you put in is a string itself.
```suggestion
        res = testapp.get(url_for(case['endpoint'], **case["kwargs"]), status=404)
```